src/Components/Components/src/Sections/SectionOutlet.cs

```diff
@@ -10,16 +10,23 @@ public sealed class SectionOutlet : ISectionContentSubscriber, IComponent, IDisp
 {
     private static readonly RenderFragment _emptyRenderFragment = _ => { };
 
-    private object? _subscribedSectionId;
+    private object? _identifier;
+    private object? _subscribedIdentifier;
     private RenderHandle _renderHandle;
     private SectionRegistry _registry = default!;
     private RenderFragment? _content;
 
     /// <summary>
-    /// Gets or sets the ID that determines which <see cref="SectionContent"/> instances will provide
+    /// Gets or sets the ID of type <see cref="string"/> that determines which <see cref="SectionContent"/> instances will provide
     /// content to this instance.
     /// </summary>
-    [Parameter, EditorRequired] public object SectionId { get; set; } = default!;
+    [Parameter] public string SectionName { get; set; } = default!;
+
+    /// <summary>
+    /// Gets or sets the ID of type <see cref="object"/> that determines which <see cref="SectionContent"/> instances will provide
+    /// content to this instance.
+    /// </summary>
+    [Parameter] public object SectionId { get; set; } = default!;
```

Comment:
I wonder if it would be worth explicitly calling out the fact that `SectionName` is effectively an alias for `SectionId`, and only one should ever be specified. I realize you added the exception message to catch this case, but an updated comment might help minimize upfront confusion. What do you think?

Reply:
For example, maybe the comment for `SectionName` could be:

```csharp
/// <summary>
/// Gets or sets the <see cref="string"/> ID of type that determines which <see cref="SectionContent"/> instances will provide
/// content to this instance.
/// </summary>
/// <remarks>
/// Only one of <see cref="SectionName"/> and <see cref="SectionId"/> can be specified.
/// </remarks>
[Parameter] public string SectionName { get; set; } = default!;
```

The `SectionId` comment could be similar.